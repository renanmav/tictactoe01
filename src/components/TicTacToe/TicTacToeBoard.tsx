import React from 'react';
import { View } from 'react-native';
import { t } from 'react-native-tailwindcss';

import TicTacToeField from './TicTacToeField';

const s = {
  board: [t.bgBlue800, t.wFull, t.rounded, t.flexRow, t.flexWrap, t.p1, { aspectRatio: 1 }],
};

interface TicTacToeBoardProps {
  board: string;
  onPress(index: number): void;
}

const TicTacToeBoard: React.FC<TicTacToeBoardProps> = ({ board, onPress }) => {
  if (board.length !== 9) {
    throw new Error('Board should have 9 characters');
  }

  return (
    <View style={s.board}>
      {board.split('').map((symbol, index) => (
        <TicTacToeField
          key={index}
          symbol={symbol as any}
          onPress={() => {
            onPress(index);
          }}
        />
      ))}
    </View>
  );
};

export default TicTacToeBoard;
