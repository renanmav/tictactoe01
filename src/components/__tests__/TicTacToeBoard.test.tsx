import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import TicTacToeBoard from '../TicTacToe/TicTacToeBoard';

it('should render the board correctly', async () => {
  const { queryAllByTestId } = render(<TicTacToeBoard board="X O     X" onPress={jest.fn} />);

  expect(queryAllByTestId('X')).toMatchSnapshot();
  expect(queryAllByTestId('X').length).toBe(2);
  expect(queryAllByTestId('O')).toMatchSnapshot();
  expect(queryAllByTestId('O').length).toBe(1);
});

it('should call on press with index when pressed', async () => {
  const onPress = jest.fn();
  const index = 3;

  const { queryAllByTestId } = render(<TicTacToeBoard board="X O     X" onPress={onPress} />);

  fireEvent.press(queryAllByTestId('field-touchable')[index]);

  expect(onPress).toBeCalledTimes(1);
  expect(onPress).toBeCalledWith(index);
});
