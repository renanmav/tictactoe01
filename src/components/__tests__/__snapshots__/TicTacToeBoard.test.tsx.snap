// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the board correctly 1`] = `
Array [
  <Image
    source={1}
    style={
      Array [
        Object {
          "width": "66.666667%",
        },
        Object {
          "resizeMode": "contain",
        },
      ]
    }
    testID="X"
  />,
  <Image
    source={1}
    style={
      Array [
        Object {
          "width": "66.666667%",
        },
        Object {
          "resizeMode": "contain",
        },
      ]
    }
    testID="X"
  />,
]
`;

exports[`should render the board correctly 2`] = `
Array [
  <Image
    source={1}
    style={
      Array [
        Object {
          "width": "66.666667%",
        },
        Object {
          "resizeMode": "contain",
        },
      ]
    }
    testID="O"
  />,
]
`;
